gateway/api/management/commands/schedule_queued_jobs.py

```diff
@@ -67,6 +70,38 @@ def handle(self, *args, **options):
                 tracer = trace.get_tracer("scheduler.tracer")
                 with tracer.start_as_current_span("scheduler.handle", context=ctx):
                     job = execute_job(job)
-                    logger.info("Executing %s", job)
+                    job_id = job.id
+                    backup_status = job.status
+                    backup_logs = job.logs
+                    backup_resource = job.compute_resource
+                    backup_ray_job_id = job.ray_job_id
+
+                    succeed = False
+                    attempts = settings.RAY_SETUP_MAX_RETRIES
+
+                    while not succeed and attempts > 0:
+                        attempts -= 1
+
+                        try:
+                            job.save()
+                            # remove artifact after successful submission and save
+                            if os.path.exists(job.program.artifact.path):
+                                os.remove(job.program.artifact.path)
 
+                            succeed = True
+                        except RecordModifiedError:
```

Comment:
Sorry for silly questions.  I'm still trying to understand.  Does this exception happen when the `update_job_statuses` update the `job'?

Reply:
yes, those exception might happen anywhere where we have `job.save()`

Reply:
I don't see it ideal but by now I understand the fix at least. But yeah, I would like to see a TODO to improve it or something.